warehouse/transformer/events.go

```diff
@@ -223,6 +212,38 @@ func (t *Transformer) extractEvents(tec *transformEventContext) ([]map[string]an
 	return []map[string]any{extractOutput}, nil
 }
 
+func (t *Transformer) extractCommonProps(tec *transformEventContext) (map[string]any, map[string]string, error) {
+	commonData := make(map[string]any)
+	commonMetadata := make(map[string]string)
+
+	if err := setDataAndMetadataFromInput(tec, t.eventContext(tec), commonData, commonMetadata, &prefixInfo{
+		completePrefix: "extract_context_",
+		completeLevel:  2,
+		prefix:         "context_",
+	}); err != nil {
+		return nil, nil, fmt.Errorf("extract: setting data and column types from input: %w", err)
+	}
+
+	eventColName, err := safeColumnNameCached(tec, "event")
+	if err != nil {
+		return nil, nil, fmt.Errorf("extract: safe column name: %w", err)
+	}
+
+	eventName, _ := tec.event.Message[eventColName].(string)
+	commonData[eventColName] = transformTableNameCached(tec, eventName)
+	commonMetadata[eventColName] = model.StringDataType
+
+	if err = setDataAndMetadataFromRules(tec, commonData, commonMetadata, rules.ExtractRules); err != nil {
+		return nil, nil, fmt.Errorf("extract: setting data and column types from rules: %w", err)
+	}
+
+	eventName, _ = commonData[eventColName].(string)
+	if utils.IsEmptyString(eventName) {
```

Comment:
Shouldn't we have this check before `setDataAndMetadataFromRules`?

Reply:
setDataAndMetadataFromRules populated eventColName, so we need to do this check post that.